### PR TITLE
Changed description for fd_t::fd in sys/include/fd.h

### LINKED ISSUE #1759: posix: fd_t struct require redundant information
Looking at `sys/include/fd.h` it seems that `fd_t` stores the pid itself **and** requires it as a paramater for the function pointers. Is there a reasoning behind? And shouldn't this struct also include a `close` function pointer?

### PR DESCRIPTION
**Pull request by kushalsingh007**
-Fixes  #1759 
-Also fixed a minor typo in previous description

### CODE REVIEW

**Comment by kushalsingh007** [comment]
@OlegHahm : Can you elaborate your comment about open function call in the issue number 1759.

**Comment by kushalsingh007** [comment]
Kindly Restart the build for this one.

**Comment by kushalsingh007** [comment]
@gebart : Can you have a look at it . This one is a fix for #1759

File: examples/ccn-lite-client/HOWTO [hunk 1]
```diff
@@ -4,7 +4,7 @@ simple appserver (all in one shell)
 0. create tap devices: *./cpu/native/tapsetup.sh create 3*
 1. build ccn-lite-client: *make -B clean all-valgrind*
 2. start: *./bin/native/ccn-lite-client.elf tap0* (valgrind support included)
-3. optinonal: *config 20* [enter] (this sets the content store size)
+3. optional: *config 20* [enter] (this sets the content store size)
```
Comment (OlegHahm) [inline examples/ccn-lite-client/HOWTO]: unrelated, please make a separate PR (or at least a separate commit)

**Comment by kushalsingh007** [comment]
Ok , apart from them is alright ?

File: sys/include/fd.h [hunk 2]
```diff
@@ -36,11 +36,11 @@ typedef struct {
     /** private status of this fd_t */
     int __active;
 
-    /** the internal filedescriptor */
-    int fd;
+    /** Stores the RIOT internal value for the file descriptor (not POSIX). */
+    int internal_fd;
```
Comment (OlegHahm) [inline sys/include/fd.h]: maybe just `__fd` - @authmillenon, any objections?

**Comment by kushalsingh007** [comment]
I created a seperate commit for the small typo fix.

File: sys/include/fd.h (hunk 2 above)
Comment (authmillenon) [inline sys/include/fd.h]: What was it with underscores? (https://github.com/RIOT-OS/RIOT/pull/2355#discussion_r23561946):

> Leading underscores are not really recommended in include guard macro names.
> 
> http://www.open-std.org/jtc1/sc22/wg14/www/docs/n1570.pdf
>
>> 7.1.3 Reserved identifiers
>> All identifiers that begin with an underscore and either an uppercase letter or another underscore are always reserved for any use.

Comment (authmillenon) [inline sys/include/fd.h]: But as far as I see `_fd` would not be reserved.

Comment (OlegHahm) [inline sys/include/fd.h]: Just thought because the first member is also prefixed this way.

Comment (authmillenon) [inline sys/include/fd.h]: Maybe @kushalsingh007 is so kind to change this too :-)

**Comment by kushalsingh007** [comment]
Done :) @authmillenon

**Comment by OlegHahm** [comment]
I guess @authmillenon meant it the other way around: replace the leading underscores with `internal_` for the first member.

**Comment by kushalsingh007** [comment]
So the orignal internal_fd is what we want right ?

**Comment by OlegHahm** [comment]
Yes, plus `internal_active`. And following our contributor guidelines add reactions to the review as additional commits and squash only after the review is finished.

**Comment by kushalsingh007** [comment]
Added a new commit with ```fd_t::internal_active``` and ```fd_t::internal_fd``` as the fields.
 @authmillenon : Done now ;)

**Comment by authmillenon** [comment]
No `::_fd` and `::_active` is alright, but okay :D

**Comment by OlegHahm** [comment]
lol - sorry for the confusion then.

**Comment by authmillenon** [comment]
(according to the standard `_[a-z].*` is okay, but `_[_A-Z].*` is not okay for identifiers ;-) and I like underscores for private variables :-))

**Comment by kushalsingh007** [comment]
What identifier were you talking about @authmillenon .
I'm a bit confused at this point.
* Should I add another commit with ```fd_t::_fd``` and ```fd_t::_active``` as the fields.
* Or squash the current commits if that's alright ?

**Comment by authmillenon** [comment]
No, it's alright, since this is a matter of taste.

> What identifier were you talking about @authmillenon .

Identifiers (generic term for variable, function, or macro names ;-)) in general. It was about the double underscore proposition of @OlegHahm. Those are not allowed in standard C, since they shall be used by the compiler to serve magic macros like `__FILE__` or `__LINE__` ;-).

Anyway, I give my ACK for this. Please squash (but keep the CCN one separate ;-)).

**Comment by kushalsingh007** [comment]
Kindly restart the build when you have time.

**Comment by authmillenon** [comment]
You must be able to restart it yourself, when you log into Travis with your Github account. But I'll do it since it's just a click ;-)

**Comment by authmillenon** [comment]
One final remark though: your commit subject is way too long. Use e.g. `fd: Clean-up documentation and rename members`

**Comment by kushalsingh007** [comment]
>You must be able to restart it yourself, when you log into Travis with your Github account. 

Do I have to enable something / have some administrative privileges in order to restart the build , because I tried clicking the button several times but nothing happened. From what I think I one should have administrative privilege for RIOT-OS/RIOT repository.  

Btw, I changed the commit subject :)

**Comment by OlegHahm** [comment]
ACK and go.
